### ISSUE TITLE
Look into using Flatpak for unified packaging

### ISSUE THREAD

**Issue by alex-ter** [tomboy-notes/tomboy#55]
As suggested by @xdevnullx, Flatpak may be a solution for app dependencies. At the first glance, looks indeed interesting and they have MonoDevelop among their apps - so it works for Mono-based ones.

This item is for tracking the work on this. If anyone is interested in helping out with this - feel free to post a comment here and go ahead!

**Comment by mlncn** [comment]
To clarify, will this solve (or help solve) the problem that the [download instructions](https://wiki.gnome.org/Apps/Tomboy/Download) for GNU/Linux state: "There is a good chance that your Linux distribution includes Tomboy. Refer to your distribution's software installation instructions."  But for at least Debian, including Debian Mint, the latest version available through the distribution is [1.15.4-1](https://packages.debian.org/experimental/tomboy), five versions behind?

**Comment by davidbannon** [comment]
Noting Ben's pretty reasonable complaint, what about, as a short term 
part fix, we (I ?) make at least some Debs every time you push out a new 
release ?  Compiling tomboy is reasonably easy but getting the right 
compile env is not.

Debian and Ubuntu (et el) users could then grab the deb and install from 
there. And maybe (just maybe) distro maintainers may be keener to pick 
up a new release if much of the work is already done.

I guess we could also do an RPM ......

We'd need (eg) for Ubuntu, current release, previous release, current 2 
long termers, 32 and 64 bit ?

David

**Comment by alex-ter** [comment]
Yeah, Flatpak advantages are understood, but after looking a bit more into this, this isn't going to be easy as their build model essentially requires building all Mono prerequisites from sources. Not impossible, but a bit more than I initially expected. But I still like to explore this.

As for rpm/deb, as you can see it gets unwieldy very quickly and given distro/version quirks each enquires maintenance and testing, so again - time.

That wiki note was written when project was more active and even today disros indeed have it, but some in outdated versions.

I appreciate the importance of packaging for the project success, so yes, David, I think given the circumstances it's worth it to spend some time on getting at least basic packaging for two-three most popular distros done. I'll prioritize this as my next tasks, but surely pull requests are more than welcome.

**Comment by ramcq** [comment]
Flatpak now online at https://flathub.org/apps/details/org.gnome.Tomboy :)

**Comment by davidbannon** [comment]
Nice.
But I am not convinced you should be saying Tomboy is also a Mac app (and a windows install is not for the faint hearted).
Davo

**Comment by ramcq** [comment]
That description will have been lovingly copy-pasted for the past decade, from one metadata format to the next, without anyone thinking about it. :) I will just elide the list of platforms as it's of no real value to someone examining the app in the Flathub store for Linux. https://github.com/flathub/org.gnome.Tomboy/commit/124d58d3

(As an aside, last time I used it - admittedly about 5 years ago - the .msi for Windows did work entirely as intended, just full of dire warnings due to the lack of signature. Perhaps that's what you meant.)

**Comment by davidbannon** [comment]
" lovingly copy-pasted for the past decade,"
Yeah, sad that it no longer applies ! 
But what you have done is a good thing !  I see lots of complaints that Tomboy was/is an app that requires users to compile them selves.  In fact, I considered putting tomboy-ng out via flatpack too but think, maybe, its more trouble than its worth (given I make debs and RPMs available).

Anyway, I have added a link to the flatpack download to the Gnome Wiki page. (I don't have access to the Tomboy Github wiki or readme.md sadly)

Davo
